ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14157,26 +14149,28 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
             }
             wholeRecord = recordTypesMatch(dataset, keyDataset);
         }
-
         if (reuseCompare)
         {
             instance->nestedctx.addQuotedLiteral("virtual ICompare * queryKeyCompare() { return &Compare; }");
             instance->nestedctx.addQuotedLiteral("virtual IHash * queryKeyHash() { return &Hash; }");
             instance->nestedctx.addQuotedLiteral("virtual ICompare * queryRowKeyCompare() { return &Compare; }");
         }
+    }
 
-        {
-            StringBuffer flags;
-            if (wholeRecord) flags.append("|HFDwholerecord");
-            if (info.keepBest) flags.append("|DFkeepbest");
-            if (flags.length())
-                doBuildUnsignedFunction(instance->classctx, "getFlags", flags.str()+1);
-        }
+    StringBuffer flags;
+    if (info.compareAllRows) flags.append("|HDFcompareall");
+    if (info.keepLeft)       flags.append("|HDFkeepleft");
+    if (info.keepBest)       flags.append("|HDFkeepbest");
+    if (wholeRecord)         flags.append("|HDFwholerecord");
+    if ( strcmp(flags.str(),"|HDFkeepleft") )
```

Comment:
minor: a couple of unusual spaces (after ( and missing after the comma).  Also I !streq would be clearer. (IMHO)